test/disconnect-spec.js

```diff
@@ -1,83 +1,76 @@
-var producer = require('../src/index')().producer;
-var consumer = require('../src/index')().consumer;
-var utils = require('../src/modules/utils');
+const docker = require('./docker');
+const assert = require('assert');
+const producer = require('../src/index')().producer;
+const consumer = require('../src/index')().consumer;
+const utils = require('../src/modules/utils');
 
-describe('disconnections', function() {
-  before(function() {
-    this.timeout(20000);
-    return docker.start();
-  });
+/* eslint func-names: "off" */
+/* eslint prefer-arrow-callback: "off" */
+describe('disconnections', function () {
+  this.timeout(20000);
 
-  after(() => {
-    return docker.stop();
-  });
+  before(() =>
+    docker.start()
+  );
+
+  after(() => docker.stop());
 
-  describe('regular pub/sub', function() {
-    var queue = 'disco:test';
+  describe('regular pub/sub', () => {
+    const queue = 'disco:test';
 
-    it('should be able to re-register to consume messages between connection failures', function(done) {
+    it('should be able to re-register to consume messages between connection failures', (done) => {
       this.timeout(20000);
-      var counter = 0;
-      consumer.consume(queue, function() {
-        counter++;
+      let counter = 0;
+      consumer.consume(queue, () => {
+        counter += 1;
         if (counter === 50) {
           done();
         }
       })
-      .then(() => {
-        return producer.produce(queue);
-      })
+      .then(() => producer.produce(queue))
       .then(() => utils.timeoutPromise(500))
+      .then(() => { assert(counter); })
+      .then(() => docker.stop())
       .then(() => {
-        assert(counter);
-      })
-      .then(() => {
-        return docker.stop();
-      })
-      .then(() => {
-        for(var i = counter; i < 50; i++) {
+        for (let i = counter; i < 50; i += 1) {
           producer.produce(queue);
         }
       })
-      .then(() => {
-        return docker.start();
-      });
+      .then(() => docker.start());
     });
   });
 
-  describe('RPC', function() {
-    var queue = 'disco:test:2';
+  describe('RPC', () => {
+    const queue = 'disco:test:2';
 
-    it('should be able to re-register to consume messages between connection failures', function(done) {
+    it('should be able to re-register to consume messages between connection failures', (done) => {
       this.timeout(20000);
-      var counter = 0;
-      var checkReceived = function(counter) {
-        if (counter === 50) done();
+      let counter = 0;
+      const checkReceived = (cnt) => {
+        if (cnt === 50) done();
       };
 
-      consumer.consume(queue, function() {
-        counter++;
+      consumer.consume(queue, () => {
+        counter += 1;
         return counter;
       })
       .then(() => {
-        return producer.produce(queue, undefined, { rpc: true }).then(checkReceived);
+        producer.produce(queue, undefined, { rpc: true }).then(checkReceived);
```

Comment:
this should still be returned. I think you need to use it without { } so it gets returned:
.then(() =>  producer.produce(queue, undefined, { rpc: true }).then(checkReceived);